### PR TITLE
Support translation of block names with periods for i18n - Test

### PR DESCRIPTION
**Pull request by nkiruka**
This PR tests that option keys with periods are translated.

### CODE REVIEW

**Comment by nkiruka** [comment]
@Hamms Not beautiful yet, but post HOC it will get there!

**Comment by nkiruka** [comment]
@Hamms Ready for review.  Thanks

File: dashboard/test/models/blockly_test.rb [hunk 1]
```diff
@@ -143,6 +143,23 @@ def assert_equal_xml(a, b)
     assert_equal xml, xml2
   end
 
+  def create_test_data(name, pool, block_text, args, category: 'Events')
```
Comment (Hamms) [inline dashboard/test/models/blockly_test.rb]: this should have a more-specific name; "test data" is extremely vague, particularly in a test file that contains such a variety of tests.

File: dashboard/test/models/blockly_test.rb [hunk 2]
```diff
@@ -143,6 +143,23 @@ def assert_equal_xml(a, b)
...
+    [{
+      name: name,
+      pool: pool,
+      category: category,
+      config:
+      {
+        "color" => [140, 1, 0.74],
+        "func" => "atSelectColor",
```
Comment (Hamms) [inline dashboard/test/models/blockly_test.rb]: why this default?

File: dashboard/test/models/blockly_test.rb [hunk 3]
```diff
@@ -412,6 +429,52 @@ def assert_equal_xml(a, b)
     assert_not_equal localized_custom_block, translated_block
   end
 
+  test 'option that contains a period in the key is translated' do
+    test_locale = :"fr-ST"
```
Comment (Hamms) [inline dashboard/test/models/blockly_test.rb]: nit: we generally use `te-ST` for testing, to ensure that we're not using a real locale

File: dashboard/test/models/blockly_test.rb [hunk 4]
```diff
@@ -412,6 +429,52 @@ def assert_equal_xml(a, b)
...
+    I18n.locale = test_locale
+    custom_i18n = {
+      "data" => {
+        "blocks" => {
+          "DanceLab_atSelectQuantity" => {
```
Comment (Hamms) [inline dashboard/test/models/blockly_test.rb]: I'd recommend not using an existing block, but instead making up a block definition just for this test

File: dashboard/test/models/blockly_test.rb (hunk 2 above)
Comment (nkiruka) [reply dashboard/test/models/blockly_test.rb]: Good catch!  Thanks.  Modified the object to only include key/values that will be used.

**Comment by nkiruka** [comment]
@Hamms Ready for review.  Thanks

File: dashboard/test/models/blockly_test.rb [hunk 5]
```diff
@@ -412,6 +426,50 @@ def assert_equal_xml(a, b)
     assert_not_equal localized_custom_block, translated_block
   end
 
+  test 'option that contains a period in the key is translated' do
+    test_locale = :"te-ST"
+    I18n.locale = test_locale
+    custom_i18n = {
+      "data" => {
+        "blocks" => {
+          "ThunderCats_atSelectStrengthLevel" => {
```
Comment (Hamms) [inline dashboard/test/models/blockly_test.rb:435]: love it!

**Review by Hamms — APPROVED** [review]

**Pull request merged**